homeassistant/components/tractive/switch.py

```diff
@@ -0,0 +1,173 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Trackables
+from .const import (
+    ATTR_BUZZER,
+    ATTR_LED,
+    ATTR_LIVE_TRACKING,
+    CLIENT,
+    DOMAIN,
+    SERVER_UNAVAILABLE,
+    TRACKABLES,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+)
+from .entity import TractiveEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class TractiveRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    method: Literal["async_set_buzzer", "async_set_led", "async_set_live_tracking"]
+
+
+@dataclass
+class TractiveSwitchEntityDescription(
+    SwitchEntityDescription, TractiveRequiredKeysMixin
+):
+    """Class describing Tractive switch entities."""
+
+
+SWITCH_TYPES: tuple[TractiveSwitchEntityDescription, ...] = (
+    TractiveSwitchEntityDescription(
+        key=ATTR_BUZZER,
+        name="Tracker Buzzer",
+        icon="mdi:volume-high",
+        method="async_set_buzzer",
+    ),
+    TractiveSwitchEntityDescription(
+        key=ATTR_LED,
+        name="Tracker LED",
+        icon="mdi:led-on",
+        method="async_set_led",
+    ),
+    TractiveSwitchEntityDescription(
+        key=ATTR_LIVE_TRACKING,
+        name="Live Tracking",
+        icon="mdi:map-marker-path",
+        method="async_set_live_tracking",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tractive switches."""
+    client = hass.data[DOMAIN][entry.entry_id][CLIENT]
+    trackables = hass.data[DOMAIN][entry.entry_id][TRACKABLES]
+
+    entities = [
+        TractiveSwitch(client.user_id, item, description)
+        for description in SWITCH_TYPES
+        for item in trackables
+    ]
+
+    async_add_entities(entities)
+
+
+class TractiveSwitch(TractiveEntity, SwitchEntity):
+    """Tractive switch."""
+
+    entity_description: TractiveSwitchEntityDescription
+
+    def __init__(
+        self,
+        user_id: str,
+        item: Trackables,
+        description: TractiveSwitchEntityDescription,
+    ) -> None:
+        """Initialize switch entity."""
+        super().__init__(user_id, item.trackable, item.tracker_details)
+
+        self._attr_name = f"{item.trackable['details']['name']} {description.name}"
+        self._attr_unique_id = f"{item.trackable['_id']}_{description.key}"
+        self._attr_available = False
+        self._tracker = item.tracker
+        self._method = getattr(self, description.method)
+        self.entity_description = description
+
+    @callback
+    def handle_server_unavailable(self) -> None:
+        """Handle server unavailable."""
+        self._attr_available = False
+        self.async_write_ha_state()
+
+    @callback
+    def handle_hardware_status_update(self, event: dict[str, Any]) -> None:
+        """Handle hardware status update."""
+        if (_state := event[self.entity_description.key]) is None:
+            return
+        self._attr_is_on = _state
```

Comment:
```suggestion
        if (state := event[self.entity_description.key]) is None:
            return
        self._attr_is_on = state
```

I wouldn't use underscores here. That quickly seems like a private / protected instance variable.

Reply:
Yes, you're right, underscore is't needed here.